go/ethclient/interface.go

```diff
@@ -23,9 +23,10 @@ type EthClient interface {
 	IsBlockAncestor(block *types.Block, proof obscurocommon.L1RootHash) bool
 
 	RPCBlockchainFeed() []*types.Block                             // returns all blocks from genesis to head
-	BroadcastTx(t *obscurocommon.L1TxData)                         // issues an obscurocommon.L1TxData to the L1 network
+	BroadcastTx(tx types.TxData)                                   // issues an obscurocommon.L1Transaction to the L1 network
```

Comment:
the comment is not accurate.

Reply:
Cleared up those different methods and shifted away responsibilities 👍